search/includes/classes/class-search.php

```diff
@@ -237,4 +240,44 @@ static function ep_skip_query_integration( $skip ) {
 		// The filter is checking if we should _skip_ query integration
 		return ! ( $query_integration_enabled || $query_integration_enabled_legacy );
 	}
+
+	/**
+	 * Filter for ep_pre_request_host
+	 *
+	 * Return the next host in our enpoint list if it's defined. Otherwise, return the last host.
+	 */
+	static function filter__ep_pre_request_host( $host, $failures, $path, $args ) {
+		if( ! defined( 'VIP_ELASTICSEARCH_ENDPOINTS' ) ) { 
+			return $host;
+		}
+
+		return self::round_robin_hosts( $host, VIP_ELASTICSEARCH_ENDPOINTS );
+	}
+
+	/**
+	 * Return the next host in the list based on the current host
+	 */
+	static function round_robin_hosts( $host, $hostList ) {
```

Comment:
I would suggest naming the functions with the action that they perform for clarity...so this could be something along these lines:

```suggestion
	static function get_next_host_from_list( $host, $hostList ) {
```